### PR TITLE
feat(rust): interactive ockam node start

### LINKED ISSUE #6659: Make `ockam node start` (no args) interactive by asking the user to choose from a list of nodes to start (tuify)
# Current behavior

When a user runs `ockam node start` without any arguments, currently they are shown help text. Here's a screenshot of this.

Image

# Desired behavior

Change the behavior of this command so that it becomes interactive with the user when no arguments are passed. Note that this can only be allowed to happen if:

- the terminal is interactive, and,
- the [`--no-input`](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/terminal/mod.rs#L265) or `--quiet` flags have *not* been passed, and,
- no arguments have been passed in to the command.

This what the user flow should look like in interactive mode.

1) Use the [tuify](https://crates.io/crates/r3bl_tuify) crate's `select_from_list()` function to ask the user to select from a list of nodes that have not been started. They should be able to make multiple selections. If there are none, then simply display a message saying that none exist (not an error) and exit.

2) Once the user has made their selection(s), display a confirmation prompt letting the user know that that `start` is about to be called with the nodes they have selected. Use `select_from_list()` to display this prompt. They should be able to make a single selection.
   - If they have made no selections then exit by displaying a message to the user letting them know they did not make any selection.

3) If they choose "YES", then run `start` for each selected nodes, and display the results of the `start` command for each nodes that they've selected.
   - If an operation succeeds then use the ✅ emoji as a prefix for the message. 
   - And if it fails, then use the ⚠️ emoji as a prefix for the message.

> The following is a video showing a generic user flow that shows the user experience when a user is asked to choose a from a list of "things". Then an action is performed on each "thing". Finally status information is reported for each action that is performed on each "thing". Please note that this flow is *not* specific to this issue. 
> Image
>  https://asciinema.org/a/612547

# Implementation details

- Please use the functions `select_multiple`, and `confirm_interactively` in this [file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/terminal/mod.rs#L281).

- Here's a similar PR to get some ideas from: https://github.com/build-trust/ockam/pull/6480

- You can learn more about how to use the `select_from_list()` in the tuify crate [here](https://docs.rs/r3bl_tuify/latest/r3bl_tuify/). Here is an [example](https://github.com/r3bl-org/r3bl_rs_utils/blob/ockam-demo/tuify/examples/main_interactive.rs) that is very similar to the steps you might have to take for this issue.
- Here's a [file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/terminal/mod.rs#L255) you can look at to detect when a terminal is interactive or not.

- Here's a [file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/node/start.rs#L41) you can look at to see where the `ockam node start` command is implemented.

---
<i>

We love helping new contributors! ❤️
If you have questions or need help as you explore, please join us on [Discord](https://discord.gg/bewvnm6zqS). If you're looking for other issues to contribute to, please checkout our [good first issues](https://github.com/build-trust/ockam/labels/good%20first%20issue).
</i>

### PR DESCRIPTION
**Pull request by Darioazzali**
interactive selection in case no argument is given to the ockam node start. In case of single selection the behavior stays identical.

## Current behavior

The `ockam start` command with no given arguments simply print the usage help.

## Proposed changes

Now when these conditions are met:
- terminal in interactive mode
- No `--quiet` flag
- No `no-input` flag
- No node are given 
a selection prompt ask the user about what node (between the UP) should be start.
If there are no node to be started the prompt doesn't appear and the user is simply notified that all nodes are actually running.
If there are eligible node to start they appear in the selection list.
If no node are selected the user is simply informed about that and exit with no error.
If there are node selected the user is prompted with a confirmation message.
If no is chosen, then the user is informed and the sub-command returns with no errors.
Then all selected node are started. The starting result are displayed to user with a tick:
```sh
    ⚠️ {node-name} wont start
     ✔︎ {node-name} 
```
This would be an example with the user that selected only one node in the selection.
Listing the nodes
node_list
User prompt selection
selection
Confirmation
confirm
Output
output
No node selected
no_selected
No available nodes to start
no_node_to_start

 
## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [x] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam/edit/develop/.github/CONTRIBUTORS.csv) file in the github web UI and create a separate Pull Request, this will mark the commit as verified.

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@Darioazzali while we still have to review this change. I'm adding the `HACKTOBERFEST-ACCEPTED` label so the contribution is counted by hacktoberfest bots before the end of October. Thank you for taking time to contribute 🎉

**Review by adrianbenavides — COMMENTED** [review]
Hi @Darioazzali thanks for the PR! We are refactoring several commands to add the `tui` experience and we are trying to keep the code as similar as possible amongst the different commands.

We recently merged the `tui` changes for the `node show` command, which uses a more explicit way of handling what items the user selected to act upon. Check it out here https://github.com/build-trust/ockam/pull/6665

**Comment by Darioazzali** [comment]
Thanks for the suggestion Adrian. A couple things to ask I'm not sure I've understood:
When you are saying that there is a more explicit way of handling what items the user selected to act upon i thing i don't grasp what are you specifically talking about.  
1. Is this command to get all the nodes name? 
`opts.state.nodes.list_items_names()?;`
used in the *show* PR?
Because the case studies It's subtle different, if I understand correctly I have to filter between the node state to be show in the list, in order to the user to select only the inactive nodes.
So there is the possibility that: 
a) All the nodes are actually started.
b) There are no nodes (in whatever state)
Should i treat them in the same manner?
2. Or you're talking about a big match with the arms calling smaller functions for readability concern?

In other words, what is the explicit way that you are mentioning?
Sorry for being verbose but I want to understand how to refactor in a similar manner despite the fact that they basically do different things. 
Thanks in advance

**Comment by adrianbenavides** [comment]
> 2\. Or you're talking about a big match with the arms calling smaller functions for readability concern?

This ☝️ What I meant is that, using that structure is more readable, which helps to identify all the possible branches that come up from the user selecting 0,1,N items, showing the full details of 1 or the N items in a list, etc.

**Comment by Darioazzali** [comment]
Thanks Adrian, got it!

**Comment by Darioazzali** [comment]
Refactored in order to give the same structure

**Review by adrianbenavides — APPROVED** [review]
Love it, looks perfect 👍  Thank you so much @Darioazzali for your contribution, will get merged soon! 🚀

**Review by etorreborre — APPROVED** [review]
